sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredentialOptions can be used when providing additional credential information, such as a client secret.
+// Also use these options to modify the default pipeline behavior through the TokenCredentialOptions.
+type InteractiveBrowserCredentialOptions struct {
+	// The Azure Active Directory tenant (directory) ID of the service principal.
+	TenantID string
+	// The client (application) ID of the service principal.
+	ClientID string
+	// The client secret that was generated for the App Registration used to authenticate the client. Only applies for web apps.
+	ClientSecret *string
+	// The redirect URI used to request the authorization code. Must be the same URI that is configured for the App Registration.
+	RedirectURI *string
+	// Options allows configuring the management of the requests sent to Azure Active Directory.
+	Options *TokenCredentialOptions
+}
+
+// InteractiveBrowserCredential enables authentication to Azure Active Directory using an interactive browser to log in.
+type InteractiveBrowserCredential struct {
+	client *aadIdentityClient
+	// options contains data necessary to authenticate through an interactive browser window
+	options InteractiveBrowserCredentialOptions
+}
+
+// DefaultInteractiveBrowserCredentialOptions use this function to get the default values for logging in with
+// an interactive browser window.
+func DefaultInteractiveBrowserCredentialOptions() InteractiveBrowserCredentialOptions {
+	return InteractiveBrowserCredentialOptions{
+		TenantID:     organizationsTenantID,
+		ClientID:     developerSignOnClientID,
+		ClientSecret: nil,
+		RedirectURI:  nil,
+	}
+}
+
+// NewInteractiveBrowserCredential constructs a new InteractiveBrowserCredential with the details needed to authenticate against Azure Active Directory through an interactive browser window.
+// options: allow to configure the management of the requests sent to Azure Active Directory, pass in nil for default behavior.
+func NewInteractiveBrowserCredential(options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
+	if options == nil {
+		*options = DefaultInteractiveBrowserCredentialOptions()
```

Comment:
This will panic if `options` is `nil`.  You'll need to capture the return value in a local then take its address.

Reply:
Probably need a test to cover this case.

Reply:
Done